src/components/Gallery.tsx

```diff
@@ -0,0 +1,101 @@
+import ImgixAPI, { APIError } from 'imgix-management-js';
+import { Component } from 'react';
+import Imgix from 'react-imgix';
+import { SourceProps } from './Dialog';
+
+interface GalleryProps {
+  selectedSource: Partial<SourceProps>;
+  imgix: ImgixAPI;
+}
+
+interface GalleryState {
+  imgix: ImgixAPI;
+  fullUrls: Array<string>;
+}
+
+export default class Gallery extends Component<GalleryProps, GalleryState> {
+  constructor(props: GalleryProps) {
+    super(props);
+
+    this.state = {
+      imgix: props.imgix,
+      fullUrls: [],
+    };
+  }
+
+  getImages = async () => {
+    return await this.state.imgix.request(
+      `assets/${this.props.selectedSource?.id}`,
+    );
+  };
+
+  getImagePaths = async () => {
+    let images,
+      allOriginPaths: string[] = [];
+
+    try {
+      images = await this.getImages();
+    } catch (error) {
+      // APIError will emit more helpful data for debugging
+      if (error instanceof APIError) {
+        console.error(error.toString());
+      } else {
+        console.error(error);
+      }
+      return allOriginPaths;
+    }
+
+    /*
+     * Resolved requests can either return an array of objects or a single
+     * object via the `data` top-level field. When parsing all enabled sources,
+     * both possibilities must be accounted for.
+     */
+    const imagesArray = Array.isArray(images.data)
+      ? images.data
+      : [images.data];
+    imagesArray.map((image: any) =>
+      // TODO: add more explicit types for image
+      allOriginPaths.push(image.attributes.origin_path),
+    );
+
+    return allOriginPaths;
+  };
+
+  /*
+   * Constructs an array of imgix image URL from the selected source in the
+   * application Dialog component
+   */
+  constructUrl(images: string[]) {
+    const scheme = 'https://';
+    const domain = this.props.selectedSource.name;
+    const imgixDomain = '.imgix.net';
+
+    const urls = images.map(
+      (path: string) => scheme + domain + imgixDomain + path,
+    );
+    return urls;
+  }
+
+  async componentDidUpdate(prevProps: GalleryProps) {
```

Comment:
Should the hook `componentDidMount` also be used to ensure that on component load, any potential images are loaded?

Reply:
Possibly, but the component _shouldn't_ [be loaded](https://github.com/imgix/contentful/pull/24/files#diff-55cd20eda3d66087438cb07b727b8ca994490621238ee7643774cbaf1d1d8d78R142) until a Source is selected. So when the Dialog initially opens there shouldn't be any images to load because it's an empty state. Hopefully that makes sense.